### PR TITLE
Avoid re-adding the sample projects to recent projects all the time

### PR DESCRIPTION
**Pull request by nirvn**
Fixes https://github.com/opengisch/QField/issues/4174

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 16a7c3f5b70b231f6a1647a2d4c7a42243eed33b: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4176-c182130b918b528068fb0b064c681d57b4b76714-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4176-c182130b918b528068fb0b064c681d57b4b76714-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4176-c182130b918b528068fb0b064c681d57b4b76714-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4176-c182130b918b528068fb0b064c681d57b4b76714-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4176-c182130b918b528068fb0b064c681d57b4b76714-arm64-android.apk)
- [all access arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4176-c182130b918b528068fb0b064c681d57b4b76714-arm-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4176-c182130b918b528068fb0b064c681d57b4b76714-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4176-c182130b918b528068fb0b064c681d57b4b76714-x86-android.apk)
</details>

File: src/core/qgismobileapp.cpp [hunk 1]
```diff
@@ -734,9 +733,16 @@ void QgisMobileapp::readProjectFile()
     title = mProject->title().isEmpty() ? mProjectFileName : mProject->title();
   }
 
+  QList<QPair<QString, QString>> projects = recentProjects();
```
Comment (domi4484) [inline src/core/qgismobileapp.cpp:736]: Here you could use a QMap< "path", "title" >, so the path can only be unique?

**Review by domi4484 — APPROVED** [review]

**Review by domi4484 — APPROVED** [review]

**Review by domi4484 — APPROVED** [review]

File: src/core/qgismobileapp.cpp (hunk 1 above)
Comment (nirvn) [reply src/core/qgismobileapp.cpp:736]: We can't otherwise we'd lose ordering of the list (which is needed to insure most recent sits on top)

**Comment by sigsdis30** [comment]
Hi,
Will this feature be in version 2.8?

**Comment by nirvn** [comment]
@slgsdls30 , correct.

**Pull request merged**